### PR TITLE
Adjust RBAC test timing

### PR DESCRIPTION
**Pull request by jlpadilla**
Signed-off-by: Jorge Padilla <jpadilla@redhat.com>

Issue: [25264](https://github.com/stolostron/backlog/issues/25264)

### CODE REVIEW

File: tests/api/queries.test.js [hunk 1]
```diff
@@ -42,13 +42,8 @@ describe(`[P3][Sev3][${squad}] Search API - Verify results of different queries`
...
 
-    user = {
-      fullName: `system:serviceaccount:${ns}:${usr}`,
-      name: usr,
-      namespace: ns,
-      token: execSync(`oc serviceaccounts get-token ${usr} -n ${ns}`),
-    }
-    await sleep(20000) // Wait for the search index to get updated.
+    await sleep(20000) // Wait for the service account and search index to get updated.
+    user = await getUserContext(usr, ns)
```
Comment (jlpadilla) [inline tests/api/queries.test.js:46]: Attempt to resolve canary failures by adding a wait after creating the service accounts.

Comment (jlpadilla) [inline tests/api/queries.test.js:46]: Resolve canary failure by requesting the token after the sleep.

File: tests/api/rbac.test.js [hunk 2]
```diff
@@ -42,7 +34,9 @@ describe(`[P2][Sev2][${squad}] Search API: Verify RBAC`, () => {
     // - Create users and objects for this test.
     const [route] = await Promise.all([getSearchApiRoute(), execCliCmdString(setupCmds)])
     searchApiRoute = route
-  }, 20000)
+
+    await sleep(10000) // Wait for service account and the search index to get updated.
```
Comment (jlpadilla) [inline tests/api/rbac.test.js:38]: Adds sleep to resolve canary failure.

File: tests/api/rbac.test.js [hunk 3]
```diff
@@ -3,26 +3,18 @@
...
-    name: userName,
-    namespace: ns,
-    token: execSync(`oc serviceaccounts get-token ${userName} -n ${ns}`),
-  }
-}
-
 describe(`[P2][Sev2][${squad}] Search API: Verify RBAC`, () => {
   beforeAll(async () => {
-    // Usng ServiceAccounts for these tests because somfiguration is simpler.
+    // Usng ServiceAccounts for rbac tests because configuration is simpler.
```
Comment (dislbenn) [inline tests/api/rbac.test.js:17]:
```suggestion
    // Using ServiceAccounts for rbac tests because configuration is simpler.
```

Comment (dislbenn) [reply tests/api/rbac.test.js:17]: This can be updated in another PR.

Comment (jlpadilla) [reply tests/api/rbac.test.js:17]: will fix in my next PR branch to avoid having to wait for prow.

**Pull request merged**